lib/models/device.coffee

```diff
@@ -1853,6 +1859,119 @@ getDeviceModel = (deps, opts) ->
 						uuid: uuid
 		.asCallback(callback)
 
+	###*
+	# @summary Enable local mode
+	# @name enableLocalMode
+	# @public
+	# @function
+	# @memberof balena.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.device.enableLocalMode('7cf02a6');
+	#
+	# @example
+	# balena.models.device.enableLocalMode(123);
+	#
+	# @example
+	# balena.models.device.enableLocalMode('7cf02a6', function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.enableLocalMode = (uuidOrId, callback) ->
+		selectedProps = [
+			'id',
+			LOCAL_MODE_SUPPORT_PROPERTIES...
+		]
+		exports.get(uuidOrId, $select: selectedProps)
+		.then (device) ->
+			checkLocalModeSupported(device)
+			exports.configVar.set(device.id, LOCAL_MODE_ENV_VAR, '1')
+		.asCallback(callback)
+
+	###*
+	# @summary Disable local mode
+	# @name disableLocalMode
+	# @public
+	# @function
+	# @memberof balena.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.device.disableLocalMode('7cf02a6');
+	#
+	# @example
+	# balena.models.device.disableLocalMode(123);
+	#
+	# @example
+	# balena.models.device.disableLocalMode('7cf02a6', function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.disableLocalMode = (uuidOrId, callback) ->
+		exports.configVar.set(uuidOrId, LOCAL_MODE_ENV_VAR, '0')
+		.asCallback(callback)
+
+	###*
+	# @summary Check if local mode is enabled on the device
+	# @name isInLocalMode
+	# @public
+	# @function
+	# @memberof balena.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @fulfil {Boolean} - has device url
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.device.isInLocalMode('7cf02a6').then(function(isInLocalMode) {
+	# 	if (isInLocalMode) {
+	# 		console.log('The device has local mode enabled');
+	# 	}
+	# });
+	#
+	# @example
+	# balena.models.device.isInLocalMode(123).then(function(isInLocalMode) {
+	# 	if (isInLocalMode) {
+	# 		console.log('The device has local mode enabled');
+	# 	}
+	# });
+	#
+	# @example
+	# balena.models.device.isInLocalMode('7cf02a6', function(error, isInLocalMode) {
+	# 	if (error) throw error;
+	#
+	# 	if (isInLocalMode) {
+	# 		console.log('The device has local mode enabled');
+	# 	}
+	# });
+	###
+	exports.isInLocalMode = (uuidOrId, callback) ->
+		exports.configVar.get(uuidOrId, LOCAL_MODE_ENV_VAR)
+		.then (value) -> value == '1'
+		.asCallback(callback)
+
+	###*
+	# @summary Returns whether local mode is supported along with a message
```

Comment:
```suggestion
	# @summary Returns whether local mode is supported along with a message describing the reason why local mode is not supported.
```
Feel free to modify it, but it would be nice to have a description what the message is.